src/wings_dissolve.erl

```diff
@@ -51,15 +51,23 @@ complement(Fs, We) -> complement(gb_sets:to_list(Fs), We).
 dissolve_1(Faces, Complement, We0) ->
     We1 = optimistic_dissolve(Faces,Complement,We0#we{vc=undefined}),
     NewFaces = wings_we:new_items_as_ordset(face, We0, We1),
-    We2 = wings_face:delete_bad_faces(NewFaces, We1),
-    We = wings_we:rebuild(We2),
+    #we{holes=Holes0} = We2 = wings_face:delete_bad_faces(NewFaces, We1),
+    Holes = update_holes(Holes0,Faces),
+    We = wings_we:rebuild(We2#we{holes=Holes}),
     case wings_we:is_consistent(We) of
 	true ->
 	    We;
 	false ->
 	    wings_u:error_msg(?__(1,"Dissolving would cause an inconsistent object structure."))
     end.
 
+update_holes([], _) -> [];
+update_holes(Holes0, Fs) when is_list(Fs) ->
+    Holes = ordsets:intersection(Holes0, Fs),
+    Holes0--Holes;
```

Comment:
Fs is not guaranteed to be an ordset here..
Can't you just do Holes0 -- Fs ?

Reply:
I did the change you as you asked. It seems to work indeed.
(I don't remember why I used that command since in the subsequent one I used your suggestion. Anyway ...)